satpy/multiscene.py

```diff
@@ -364,7 +374,7 @@ def blend(self, blend_function=stack):
         common_datasets = self.shared_dataset_ids
         for ds_id in common_datasets:
             datasets = [scn[ds_id] for scn in self.scenes if ds_id in scn]
-            new_scn[ds_id] = blend_function(datasets)
+            new_scn[ds_id] = blend_function(datasets, **kwargs)
```

Comment:
This shouldn't be needed if `functools.partial` is used for the `blend_function`. Using `partial` you could create a callable object with the kwargs (or args) already applied I believe.

Reply:
> I'm not sure I like the final stacked result of your first example better, but I see the usefulness. It is kind of amazing how much can be done just by throwing on some weights and/or masks.

Yes, it is amazing, I agree! Satpy is fantastic!
Concerning what is better or worse in this case stacking with our without weights, is of course not important. The thing is we want to achieve the best possible cloud analysis every hour over our area, and normally (this is an assumption but can also be documented using validation data) we would get better cloud mask, type and height from the polar orbiters if we have them. Especially in the north. The sat-zenith angle is a good proxy to use here. I intend to mix in also the observation time, so that observations far away from the nominal time gets lower weight. Also, I will want to weight VIIRS cloud products the highest, higher than AVHRR. So, if you have a VIIRS cloud product at 10:45 and an AVHRR cloud product at 11:15, the 12:00 analysis will have more of VIIRS than of AVHRR, maybe even a high zenith angle for VIIRS is better than a low one for AVHRR... You can go on make it more and more complicated/sophisticated.

Reply:
> 

@djhoese I am new to `partial`. But was reading the documentation and trying to understand what you are after but couldn't make sense of it. Sorry, can you elaborate a bit further or show me what you had in mind and I can test it...?

Reply:
As far as I can tell here you are passing `**kwargs` from the user. Nothing from the keyword arguments is coming "dynamically" from the data being loaded. So a user could do something like this:

```
In [1]: from functools import partial

In [2]: def my_func(a=1, b=2):
   ...:     print(a, b)
   ...:

In [3]: pfunc = partial(my_func, a=3)

In [4]: pfunc(b=3)
3 3

In [5]: pfunc(a=2)
2 2

In [6]: pfunc(a=1)
1 2
```

The point being, by using `partial` we can **attach** args and keyword arguments to any function and then pass it as a callback to the blend function. So users could do `stack_with_weights = partial(stack, weight=my_weights)` and then `mscn.blend(blend_function=stack_with_weights)`. This is "better" in the sense that we avoid accepting `**kwargs` in the `blend` method. At the very least, if we don't want users to use `partial` we should make this a `blend_kwargs=None` keyword argument that gets passed if provided so we don't have to use the wildcard `**kwargs`.

Reply:
Thanks @djhoese I think I got it now. I have followed your suggestion.